tests/sundials/n_vector_clone_01.cc

```diff
@@ -0,0 +1,51 @@
+//-----------------------------------------------------------
+//
+//    Copyright (C) 2020 - 2021 by the deal.II authors
+//
+//    This file is part of the deal.II library.
+//
+//    The deal.II library is free software; you can use it, redistribute
+//    it, and/or modify it under the terms of the GNU Lesser General
+//    Public License as published by the Free Software Foundation; either
+//    version 2.1 of the License, or (at your option) any later version.
+//    The full text of the license can be found in the file LICENSE.md at
+//    the top level directory of deal.II.
+//
+//-----------------------------------------------------------
+
+// Test SUNDIALS' clones on our own vectors.
+
+#include <deal.II/lac/vector.h>
+
+#include <deal.II/sundials/n_vector.h>
+
+#include "../tests.h"
+
+using namespace SUNDIALS::internal;
+
+int
+main()
+{
+  initlog();
+
+  const unsigned int N = 4;
+  using VectorType     = Vector<double>;
+
+  VectorType y_dealii(N);
+  y_dealii = 1.0;
+
+  // The commented version works, the other one segfaults.
+  // auto y_sundials = make_nvector_view(y_dealii);
+  N_Vector y_sundials     = make_nvector_view(y_dealii);
```

Comment:
Problem is that, currently, one needs to keep the actual `NVectorView` object around, see also the documentation of `make_nvector_view`. In this call, the destructor of  `NVectorView` frees the actual, internally-hidden N_Vector struct right-away and the `y_sundials` is a dangling pointer. The assignment works because of the conversion operator.

What does this mistake teach us, i.e., how can we avoid this in the future? Potentially it might make sense to remove the conversion operator and force an explicit conversion (necessary e.g. in the line below)?

Reply:
Isn't it around in the `content` of the N_Vector?

Reply:
The `content` field is set to an `NVectorContent` for deal.II vectors. This object is precisely what the view object creates on construction and frees upon deletion.

Reply:
The biggest problem here is that we don't have control of how many new vectors are needed internally in sundials, and they are all created using `N_VClone` on a N_Vector object. So this:
```
auto y_sundials = make_nvector_view(y_dealii);
N_Vector clone = N_VClone(y_sundials);
N_Vector clone2 = N_VClone(clone);
```
must be supported. This should be ok, as far as I can see (see test two). 

Let me explore a bit more where we are actually making the conversion and loosing the view.

Reply:
Yes that should work.

> Let me explore a bit more where we are actually making the conversion and loosing the view.

Okay, keep me posted and/or tell me if there is anything I could do to help debug this!